diff -Naur include.ori/ADTSAudioFifoServerMediaSubsession.hh include/ADTSAudioFifoServerMediaSubsession.hh
--- include.ori/ADTSAudioFifoServerMediaSubsession.hh	2023-04-15 23:22:04.964120526 +0200
+++ include/ADTSAudioFifoServerMediaSubsession.hh	2023-04-20 19:18:14.115207781 +0200
@@ -54,6 +54,8 @@
 private:
     char* fAuxSDPLine;
     char fDoneFlag; // used when setting up "fAuxSDPLine"
+    unsigned fSamplingFrequency;
+    unsigned fNumChannels;
     char fConfigStr[5];
     RTPSink* fDummyRTPSink; // ditto
     StreamReplicator *fReplicator;
diff -Naur src.ori/ADTSAudioFifoServerMediaSubsession.cpp src/ADTSAudioFifoServerMediaSubsession.cpp
--- src.ori/ADTSAudioFifoServerMediaSubsession.cpp	2023-04-15 23:22:04.964120526 +0200
+++ src/ADTSAudioFifoServerMediaSubsession.cpp	2023-04-20 21:46:20.842280500 +0200
@@ -128,6 +128,8 @@
         Medium::close(resultSource);
         return NULL;
     } else {
+        fSamplingFrequency = originalSource->samplingFrequency();
+        fNumChannels = originalSource->numChannels();
         sprintf(fConfigStr, originalSource->configStr());
         if (debug & 4) fprintf(stderr, "createStreamReplica completed successfully\n");
 
@@ -139,10 +141,9 @@
 ::createNewRTPSink(Groupsock* rtpGroupsock,
 		   unsigned char rtpPayloadTypeIfDynamic,
 		   FramedSource* inputSource) {
-  ADTSAudioFifoSource* adtsSource = (ADTSAudioFifoSource*)inputSource;
   return MPEG4GenericRTPSink::createNew(envir(), rtpGroupsock,
 					rtpPayloadTypeIfDynamic,
-					adtsSource->samplingFrequency(),
+					fSamplingFrequency,
 					"audio", "AAC-hbr", fConfigStr,
-					adtsSource->numChannels());
+					fNumChannels);
 }
