diff -Naur wsdd.ori/src/wsdd.c wsdd/src/wsdd.c
--- wsdd.ori/src/wsdd.c	2020-08-06 20:42:04.123972145 +0200
+++ wsdd/src/wsdd.c	2020-08-06 20:37:38.000000000 +0200
@@ -340,6 +340,7 @@
     }
 
     setsockopt(soap_srv->master, IPPROTO_IP, IP_MULTICAST_IF, &mcast.imr_address.s_addr, sizeof(struct in_addr));
+    mcast.imr_ifindex = 0;
 
     if(setsockopt(soap_srv->master, IPPROTO_IP, IP_ADD_MEMBERSHIP, (char *)&mcast, sizeof(mcast)) != 0 )
     {
